src/snowflake/sqlalchemy/snowdialect.py

```diff
@@ -575,8 +716,8 @@ def _get_table_columns(self, connection, table_name, schema=None, **kw):
         ans = []
         current_database, _ = self._current_database_schema(connection, **kw)
         full_schema_name = self._denormalize_quote_join(current_database, schema)
-        schema_primary_keys = self._get_schema_primary_keys(
-            connection, full_schema_name, **kw
+        table_primary_keys = self._get_table_primary_keys(
+            connection, full_schema_name, table_name, **kw
         )
         result = connection.execute(
```

Comment:
the test fails because `_get_table_columns` lacks the handling for the `identity` attribute. This is probably a bug by us, could you help add the missing part related to `identity` into `_get_table_columns`.

just follow [`_get_schema_columns`](https://github.com/snowflakedb/snowflake-sqlalchemy/blob/72bd5ba48b7121dfd27e7fda98ea05d07bcaf668/src/snowflake/sqlalchemy/snowdialect.py#L471) implementation and you shall be good.